### PR TITLE
removing unused code/types from ingester

### PR DESCRIPTION
**Pull request by alanprot**
**What this PR does**:
Removing types/code used on the old and unused `ingester.Query` method (we still had the implementation even though it was removed from the proto file).

This method was removed here: https://github.com/cortexproject/cortex/pull/5867/files#diff-5b6c87318632a5b2830782c3245ecaffa95a2c45d49cd155050078a05fca2c05L16

Note that for the backward compatibility test we had to add the `-ingester.stream-chunks-when-using-blocks=true` on that version the default was false and we are removing the `timeseries` from the `QueryStreamResponse` - this should be ok as the default is to stream chunks since 1.14.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Tests updated
- [NA] Documentation added
- [NA] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/client/compat.go [hunk 1]
```diff
@@ -66,6 +68,28 @@ func FromExemplarQueryRequest(req *ExemplarQueryRequest) (int64, int64, [][]*lab
...
+		ss.Metric = cortexpb.FromLabelAdaptersToMetric(cortexpb.FromLabelsToLabelAdapters(s.Labels()))
+		ss.Values = make([]model.SamplePair, 0)
+		it := s.Iterator(nil)
+		for it.Next() != chunkenc.ValNone {
+			t, v := it.At()
+			ss.Values = append(ss.Values, model.SamplePair{
+				Value:     model.SampleValue(v),
+				Timestamp: model.Time(t),
+			})
+		}
```
[the remark this answers is not in the document]
Comment (yeya24) [reply pkg/ingester/client/compat.go:86]: Maybe it is fine as it is used in test only

File: pkg/querier/distributor_queryable.go [hunk 2]
```diff
@@ -147,12 +147,7 @@ func (q *distributorQuerier) streamingSelect(ctx context.Context, sortSeries boo
 		return storage.ErrSeriesSet(err)
 	}
 
-	// we should sort the series if we need to merge them even if sortSeries is not required by the querier
-	sortSeries = sortSeries || (len(results.Timeseries) > 0 && len(results.Chunkseries) > 0)
 	sets := []storage.SeriesSet(nil)
-	if len(results.Timeseries) > 0 {
-		sets = append(sets, newTimeSeriesSeriesSet(sortSeries, results.Timeseries))
```
Comment (yeya24) [inline pkg/querier/distributor_queryable.go:154]: Let's also remove `newTimeSeriesSeriesSet` and relevant data structs since they are not used anymore?

**Review by yeya24 — COMMENTED** [review]
Nit.

```
if len(serieses) > 0 {
		sets = append(sets, series.NewConcreteSeriesSet(sortSeries || len(sets) > 0, serieses))
	}
```

We can just return `series.NewConcreteSeriesSet`?

File: pkg/querier/distributor_queryable.go (hunk 2 above)
Comment (alanprot) [reply pkg/querier/distributor_queryable.go:154]: Those are being used on some tests to create promql responses. =/

**Comment by alanprot** [comment]
> Nit.
> 
> ```
> if len(serieses) > 0 {
> 		sets = append(sets, series.NewConcreteSeriesSet(sortSeries || len(sets) > 0, serieses))
> 	}
> ```
> 
> We can just return `series.NewConcreteSeriesSet`?

We already do that 2 lines below?

https://github.com/cortexproject/cortex/blob/0d0f2ad1d428afd1fc0621c7f25fb30b931f879f/pkg/querier/distributor_queryable.go#L179-L188

File: pkg/ingester/client/compat.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alanprot) [reply pkg/ingester/client/compat.go:86]: Done!

**Comment by yeya24** [comment]
> We already do that 2 lines below?

Yeah. It is a nit to cleanup some code to be more readable. Since now len(sets) cannot more than 1.

File: pkg/ingester/client/compat.go [hunk 3]
```diff
@@ -87,7 +87,7 @@ func MatrixFromSeriesSet(set storage.SeriesSet) model.Matrix {
 		m = append(m, &ss)
 	}
 
-	return m
+	return m, set.Err()
```
Comment (yeya24) [inline pkg/ingester/client/compat.go:90]: Do we need to check `it.Err()` inside the for loop?

Comment (alanprot) [reply pkg/ingester/client/compat.go:90]: Oh i see.. good catch!

Comment (alanprot) [reply pkg/ingester/client/compat.go:90]: Do we? i think if err occurred the `it.Next()` would return false, and we would go to the return line

**Comment by alanprot** [comment]
> > We already do that 2 lines below?
> 
> Yeah. It is a nit to cleanup some code to be more readable. Since now len(sets) cannot more than 1.

Good catch! :D lemme do that

File: pkg/ingester/client/compat.go (hunk 3 above)
Comment (yeya24) [reply pkg/ingester/client/compat.go:90]: Umm, will `set.Err()` return the same error as `it.Err()`?
I think `set.Err()` just returns nothing

Comment (yeya24) [reply pkg/ingester/client/compat.go:90]: I am going to merge since it is used in test only

Comment (alanprot) [reply pkg/ingester/client/compat.go:90]: No.. im fixing it

**Comment by yeya24** [comment]
I see `integration_backward_compatibility` still failing and response data mismatch for 1.13.1 @alanprot

File: pkg/ingester/client/compat.go (hunk 3 above)
Comment (alanprot) [reply pkg/ingester/client/compat.go:94]: done

**Comment by alanprot** [comment]
> I see `integration_backward_compatibility` still failing and response data mismatch for 1.13.1 @alanprot

Seems that on my rebase the change on the `TestNewDistributorsCanPushToOldIngestersWithReplication` was deleted! =/

**Review by yeya24 — APPROVED** [review]

**Pull request merged**